source/vstd/seq_lib.rs

```diff
@@ -496,6 +496,26 @@ impl<A> Seq<A> {
 
     }
 
+    /// Remove item a at index i, pop the last item and write the last item at i
+    pub open spec fn swap_remove(self, i: int) -> Seq<A>
+        recommends
+            0 <= i < self.len(),
+    {
+        self.subrange(0, i).push(self.last()) + self.subrange(i+1, (self.len() - 1) as int)
+    }
```

Comment:
I don't think this has the correct behavior for `i == self.len() - 1`.

When `i == self.len() - 1` the correct definition should just pop the last element, but in this definition we would have `self.subrange(0, i).push(self.last())` reconstructing the entire sequence, and then `self.subrange(i+1, (self.len() - 1) as int)` violating its recommends-clause.